portal/components/react-admin/base/resources/candidateProfile/candidate-show.js

```diff
@@ -0,0 +1,205 @@
+import React from "react";
+import {
+  List,
+  Datagrid,
+  Pagination,
+  TextField,
+  DateField,
+  FunctionField,
+  TopToolbar,
+  sanitizeListRestProps,
+  UrlField,
+  BooleanField,
+  RichTextField,
+  ArrayField,
+  SingleFieldList,
+  FileField,
+  Show,
+  Tab,
+  TabbedShowLayout,
+  SimpleShowLayout,
+  Filter,
+  SelectInput,
+  SearchInput,
+  ExportButton,
+  AutocompleteInput,
+  useQuery,
+  downloadCSV,
+} from "react-admin";
+import { makeStyles, Typography } from "@material-ui/core";
+
+function getAge({ start, end }) {
+  var today = end ? new Date(end) : new Date();
+  var birthDate = new Date(start);
+  var age = today.getFullYear() - birthDate.getFullYear();
+  var m = today.getMonth() - birthDate.getMonth();
+  let roundedDownAge = age;
+  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+    roundedDownAge--;
+  }
+  if (today < birthDate) {
+    return { years: "Invalid Date", months: "Invalid Date" };
+  }
+  return { years: roundedDownAge, months: age * 12 + m };
+}
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "calc(100% - 0px)",
+    height: "86vh",
+    marginTop: theme.spacing.unit * 3,
+    overflowX: "auto",
+    overflowY: "scroll",
+    marginLeft: "1rem",
+  },
+}));
+
+export const CandidateShow = (props) => {
+  const CustomFileField = ({ record, ...props }) => {
+    const url = generateResumeLink(record?.data?.[0]?.resume?.url);
+    return (
+      <div
+        style={{
+          margin: "15px 0",
+          width: "fit-content",
+          borderBottom: "1px dashed black",
+        }}
+      >
+        {url ? (
+          <a href={url} target="_blank" rel="noopener noreferrer">
+            <Typography variant="body2">Resume</Typography>
+          </a>
+        ) : (
+          <Typography variant="body2">No Resume uploaded</Typography>
+        )}
+      </div>
+    );
+  };
+  return (
+    <Show {...props} title="Candiate details">
+      <TabbedShowLayout>
+        <Tab label="Candidate Bio">
+          <TextField label="Name" source="name" />
+          <DateField label="DOB" source="DOB" />
+          <FunctionField
+            label="Age"
+            render={(record) => {
+              if (record) {
+                return getAge({
+                  start: record.DOB,
+                  end: null,
+                }).years;
+              }
+            }}
+          />
+
+          <TextField label="Gender" source="gender.gender_name" />
+          <TextField label="Mobile" source="mobile_number" />
+          <TextField label="Whatsapp" source="whatsapp_mobile_number" />
+          <TextField label="Distict" source="district_name.name" />
+          <TextField label="Pincode" source="pincode" />
+        </Tab>
+        <Tab label="Educational Details">
+          <TextField
+            label="Max Qualification"
+            source="highest_level_qualification.highest_level_qualification_name"
```

Comment:
Can we use optional chaining in this file as well?

Reply:
` source="highest_level_qualification?.highest_level_qualification_name"`
If  I use optional chaining , values are not displaying.